### DISCUSSION TITLE
Erro AssinaturaA1

### DISCUSSION THREAD

**Discussion by easy-mcaguiar** [TadaSoftware/PyNFe#322] [Q&A]
Ao executar o código, estou recebendo o erro na linha 94:
a1 = AssinaturaA1(certificado, senha, 'ginfes')

TypeError: AssinaturaA1.__init__() takes 3 positional arguments but 4 were given

Alguém já passou por esse erro? existe alguma versão mais nova ou ocorreram alterações?

**Comment by juniortada** [comment]
@easy-mcaguiar o código para assinar/emitir NFS-e está bem desatualizado, será necessário refatorar grande parte da lib.
